File: content-solutions/two-pointers/water-box.js
function solution(walls) {
  let l = 0;
  let r = walls.length - 1;
  let best = 0;
  while (l < r) {
    best = Math.max(best, Math.min(walls[l], walls[r]) * (r - l));
    if (walls[l] < walls[r]) l++;
    else r--;
  }
  return best;
}

File: content-solutions/two-pointers/water-box.py
def solution(walls):
    l, r = 0, len(walls) - 1
    best = 0
    while l < r:
        best = max(best, min(walls[l], walls[r]) * (r - l))
        if walls[l] < walls[r]:
            l += 1
        else:
            r -= 1
    return best
